### PR TITLE
Explain server config for history mode

### PR DESCRIPTION
**Pull request by iWader**
Following on from #621 I've added docs to explain server configurations when using the router's history mode.

Just looking for feedback at this point. If this is heading in the right direction I'll add more examples for other webservers.

### CODE REVIEW

File: docs/en/server.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  try_files $uri $uri/ /index.html;
+}
+```
+
+## Caveats
+
+To get around the issue that the server will no longer serve 404 errors you should implement a catch-all route within your Vue app to show a 404 page.
+
+```javascript
+router.map({
```
Comment (posva) [inline docs/en/server.md]: This is actually the docs for vue-router 2, you have to use the new syntax:

``` js
new VueRouter({
  routes: [...]
})
```

Can you use a 2 spaces indentation, please?

**Comment by posva** [comment]
Looks good as a starting point 😄 
@fnlctrl What do you think?

**Comment by LinusBorg** [comment]
This will definitely be helpful.

**Comment by fnlctrl** [comment]
@posva @LinusBorg  Looking good to me!

**Comment by iWader** [comment]
I'll add a few further config examples tonight and see where we go from
there

**Comment by posva** [comment]
@fnlctrl Merging this. @iWader you'll be able to create a new PR. You'll also be able to use the latest version

**Pull request merged**